### DISCUSSION TITLE
How to add a custom api?

### DISCUSSION THREAD

**Discussion by MamonovaMaria** [apache/superset#17791] [Q&A / Help]
Hi!

When I add a custom api (example to the dashboard class to superset\dashboards\api.py) and then write it to superset\constants.py in MODEL_VIEW_RW_METHOD_PERMISSION_MAP, I get a 404 error when accessing it.

What needs to be done in order for my custom api to be noticed by the superset?

**Comment by nytai** [comment]
you need to "register" it. see https://github.com/apache/superset/blob/4beaa81de341b055116a2d853acb193820862715/superset/initialization/__init__.py#L206

**Reply by JunTech** [reply]
i meet the same question as what you meet,i want to know how you solve it,version : superset 1.3.2,i add a custom api in dashboardRestApi,and i add the map,the flaskapp init has registered add_api(DashboardRestApi)

**Reply by raaahuuulll** [reply]
How to register it if I cannot access the __init__.py file. Like in my case I am deploying superset using Docker. So the packages will be fetched when building the image. One thing that I can think of is to register my API in the superset_config.py. But I am not sure if it is possible. Please suggest some other way where we don't need to change the files that come with the installtion.

**Reply by sughosh-kaushik-eightfold** [reply]
@raaahuuulll I want to do the same where I define the custom api in the config file and use it. But I am unable to do so. Any pointers?

**Comment by BhargabSixthSens** [comment]
Any update on this?
I want add a new api in `/superset/database/api.py` but I am getting 404 for the api
